### DISCUSSION TITLE
Xamarin.Forms communication compatibility

### DISCUSSION THREAD

**Discussion by alex6dj** [microsoft/vs-streamjsonrpc#757] [Q&A] [answered]
I wants to know if this can be used for internetwork communication betwen a Windows PC and a Xamarin.Forms (Android) in the same WiFi network. I´m already using it, for at least two years now, for inter-pc comunication, but I want to test if it can be used for Android too, for use in a remote control and notifications app. And of course if there are any limitations or requirements to use it in this way.

Thanks for this awesome piece of code.

**Comment by AArnott** [comment, marked as answer]
Yes, I would expect so. We do not test on Android nor offer support when it fails, to be clear. But there's nothing about this library that makes me believe it would fail. The protocol itself is just JSON-RPC and the library works on Windows, OSX and Linux for sure. 

If you try it and it doesn't work, I would be interested in hearing about it. But I couldn't make any promises about fixing it.

**Reply by alex6dj** [reply]
Well, just yesterday do some initial test and it seems to work almost out of the box. The only issue I have was with System.Threading.Tasks.Dataflow not being recognized correctly as a dependency. After included as a reference in my project it work correctly.

**Reply by AArnott** [reply]
Dataflow should appear as a package reference. It's a touch special in that we exclude it from your compiler, but it should be seen as a runtime dependency of the application. If you see differently, you may want to file a bug against the Xamarin tooling.
